mantidproject/mantid
# Refactor getSpectrum to return a reference with covariant return type

Make sure to merge https://github.com/mantidproject/mantid/pull/16477 before this.

Fixes #16454.

See issue for description.

**To test:**

Code review:
Mostly the changes are simple. One thing to pay special attention to:

``` cpp
auto spec = ws->getSpectrum(i);
```

Previously `auto` would be a pointer, now it will make a **copy** of the spectrum. Make sure I changed all (relevant) occurences to:

``` cpp
auto &spec = ws->getSpectrum(i);
```

Internal change, no release notes.

**Important:**

**This commit changes the interface, do not merge without warning! We should probably first inform all developers via the mailing list and maybe also in the review meeting.**

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
